### PR TITLE
Do not include arch tag in image builds

### PR DESCRIPTION
**Pull request by seemethere**
Not sure why were including them in the first place but there's that.

Signed-off-by: Eli Uriegas <eli.uriegas@docker.com>

### CODE REVIEW

File: image/Makefile [hunk 1]
```diff
@@ -66,4 +66,4 @@ engine-$(ARCH)-docker-compat.tar: image-linux
 
 .PHONY: release
 release:
-	docker push $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION).$(ARCH)
+	docker push $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION)
```
Comment (jose-bigio) [inline image/Makefile]: If the release target is being called then there should probably be a tagging step before it to push the image with the ARCH. If not I think this just ends up overwriting image  on hub.

Comment (seemethere) [reply image/Makefile]: Do we actually even use this target though?

Comment (jose-bigio) [reply image/Makefile]: I don't think we use it.

**Comment by seemethere** [comment]
PR test failure is unrelated to this PR, it was a hub EOF

**Pull request merged**